src/new/sendTransaction.ts

```diff
@@ -494,4 +509,162 @@ class SendTransaction extends EventEmitter {
   }
 }
 
-export default SendTransaction;
+/**
+ * Check the tx data and propose inputs and outputs to complete the transaction.
+ * We will only check a single token
+ *
+ * @param {IStorage} storage
+ * @param {IDataTx} dataTx
+ * @param {IUtxoSelectionOptions} options
+ */
+export async function prepareSendTokensData(
+  storage: IStorage,
+  dataTx: IDataTx,
+  options: IUtxoSelectionOptions = {},
+): Promise<Pick<IDataTx, 'inputs'|'outputs'>> {
+  const token = options.token || HATHOR_TOKEN_CONFIG.uid;
+  const utxoSelection = options.utxoSelectionMethod || bestUtxoSelection;
+  const newtxData: Pick<IDataTx, 'inputs'|'outputs'> = { inputs: [], outputs: [] };
+  let outputAmount = 0;
+
+  // Calculate balance for the token on the transaction
+  for (const output of dataTx.outputs) {
+    if (isDataOutputCreateToken(output)) {
+      // This is a mint output
+      // Since the current transaction is creating the token we can safely ignore it
+      continue;
+    }
+    const outputToken = output.token || HATHOR_TOKEN_CONFIG.uid;
+    if (outputToken !== token) {
+      // This output is not for the token we are looking for
+      continue;
+    }
+    outputAmount += output.value;
+  }
+
+  if (options.chooseInputs) {
+    if (outputAmount === 0) {
+      // We cannot process a target amount of 0 tokens.
+      throw new Error('Invalid amount of tokens to send.');
+    }
+
+    // We will choose the inputs to fill outputAmount.funds
+    const newUtxos = await utxoSelection(storage, token, outputAmount);
+    newtxData.inputs = newUtxos.utxos.map((utxo) => {
+      return {
+        txId: utxo.txId,
+        index: utxo.index,
+        value: utxo.value,
+        authorities: utxo.authorities,
+        token: utxo.token,
+        address: utxo.address,
+      } as IDataInput;
+    });
+
+    if (newUtxos.amount > outputAmount) {
+      // We need to create a change output
+      const changeAddress = options.changeAddress || await storage.getCurrentAddress();
+      const changeOutput: IDataOutput = {
+        type: 'p2pkh', //get from wallet
```

Comment:
This should be fixed, right? Otherwise will be wrong for multisig. (the same for the other change)